Create GSoC-2021.md (#405)
alan-turing-institute/AutisticaCitizenScience: project-communications/applications/GSoC-2021.md

# GSoC 2021 application summary

## Google Summer of Code (GSoC) 2021

-   Website: [https://summerofcode.withgoogle.com/](https://summerofcode.withgoogle.com/) and [https://google.github.io/gsocguides](https://google.github.io/gsocguides)     
-   Context: Google Summer of Code is a global program focused on bringing more student developers into open source software development. Students work with an open source organization on a 10-week programming project during their break from school.
- [INCF call for the organisation partners](https://neurostars.org/t/incf-call-for-community-project-proposals-and-mentors-for-gsoc-2021-deadline-for-expressions-of-interest-january-15/17865), [supporting document with details](https://docs.google.com/document/d/1T7QCv5WDxqUt1FNHiAhbE74c7e7eutzFuv3wrqToVyo/edit)
- Submission Date: 15 January 2020

### Title

* AutSPACEs: a co-created citizen science platform 

> 

Reviewer: ```suggestion

### Tech Keywords
GitHub, Python, Django, UI/UX design, web development
```

Reviewer: Moving the tech keywords to the top to avoid breaking the flow of project description below

### Project Description

> AutSPACES is a co-created, participatory, citizen-science platform which will be used to investigate how sensory processing differences affect the daily experiences of autistic people. 

Reviewer: ```suggestion
AutSPACEs is a co-created, participatory, citizen-science platform which will be used to investigate how sensory processing differences affect the daily experiences of autistic people.
```

> By collecting together diverse real-world experiences, contributed by autistic citizen scientists, we will build a dataset which can be used to make recommendations to improve autistic people's lives. 

Reviewer: ```suggestion
By collecting together diverse examples of real-world experiences contributed by autistic citizen scientists, we will build a dataset that can be used to make recommendations to improve autistic people's lives.
```

> The platform will also provide a safe and inclusive space for autistic people to share stories, strategies, and build communiity with one another. 

Reviewer: ```suggestion
The platform will also provide a safe and inclusive space to share stories, strategies, and build a community of our contributors, which will be facilitated by a community-led, co-created moderation process and code of conduct.
```

> The inclusive space will be facilitated by a community-led, co-created moderation process and code of conduct.

Reviewer: ```suggestion
```

Reviewer: Merged with the previous statement.

Supported by extensive research, and the collaborative input of autistic people and their families, we are currently building up a working prototype.

Integral to the platform is a fine-grained consent model which allows users to specify who they would like their data to be shared with, and for what purposes. 

> The platform will also grant each user dynamic control of their data. This is supported by integration with a backend infrastructure provided by [Open Humans](https://www.openhumans.org/). 

Reviewer: ```suggestion
The platform will grant each user a dynamic control of their data. 
This is supported by integration with a backend infrastructure provided by [Open Humans](https://www.openhumans.org/). 
```

> Open Humans is a foundation which enables research participants to have awareness and understanding of, as well as agency over, their own data. Supporting data agency and transparent data mangement are critical to the platform's design. 

Reviewer: ```suggestion
Open Humans is a foundation which enables research participants to have awareness and understanding of, as well as an agency over, their own data. 
Supporting data agency and transparent data management is critical to the platform's design. 
```

Reviewer: new line break

In summary, AutSPACEs enables rich, scalable, community-led research to be conducted in a way that is both transparent and encourages agency. 

> Furthermore, as an openly licensed and publicly documented research tool, it is freely available for re-use. 

Reviewer: ```suggestion
Furthermore, as an openly licensed and publicly documented research tool, others can freely re-use, build on or adapt it to investigate other research questions. 
```

> Thus, it can easily be built on or adapted to investigate other research questions. 

Reviewer: ```suggestion
```

Reviewer: Merged with the previous statement

> 
> ### Tech Keywords: GitHub, Python, Django,  

Reviewer: ```suggestion
```

Reviewer: Moved this to the top

### Technical details

The platform is being built openly on GitHub. 
Data, documentation, details, and community recommendations, as well as supporting documents, protocols, and more, are all publicly available on our [project management repository](https://github.com/alan-turing-institute/AutisticaCitizenScience). 

> The platform development takes place in a separate, linked repository: [AutSPACEs](https://github.com/GeorgiaHCA/AutSPACEsEvery) stage of the project is being conducted in collaboration with autistic people, and we are committed to being open and transparent throughout. The platform is being developed using the Python/Django web development framework. 

Reviewer: ```suggestion
The platform development takes place in a separate, linked repository: [AutSPACEs](https://github.com/GeorgiaHCA/AutSPACEsEvery) using the Python/Django web development framework.
```

Reviewer: Moving half the sentence to the end of this paragraph

> Users will be able to share experiences through a website which will then be deposited into a backend database supported by Open Humans.

Reviewer: ```suggestion
Users will be able to share experiences through a website, which will then be deposited into a backend database supported by Open Humans.
All stages of the project are being conducted in collaboration with autistic people, and we are committed to being open and transparent throughout. 
```

### GSoC project plan

We are seeking a GSOC mentee with web development skills to work on building the platform with a researcher and an experienced developer. 
This will involve elements of UX/UI design focussing on the needs and preferences of autistic collaborators. 
We are able to offer a unique opportunity for a GSOC mentee to grow their technical proficiency and collaborative skills in the context of innovative, impact-focussed research. 
We will provide our mentee with a firm grounding in the emerging best practices of open source development and research. 

> They will also be welcomed into the [The Turing Way](https://www.turing.ac.uk/research/research-projects/turing-way-handbook-reproducible-data-science) community. 
> Depending on their preferences and interests, they will be invited to join the _The Turing Way_ community who are building an open source book to improve data science practices, to work with autistic collaborators on participatory design, and to help promote the platform to onboard new developers. 

Reviewer: ```suggestion
Depending on their preferences and interests, they will be invited to join the [_The Turing Way_](https://www.turing.ac.uk/research/research-projects/turing-way-handbook-reproducible-data-science) - an open handbook on data science, to work with autistic collaborators on participatory design, and to help promote the platform to onboard new developers. 
```

### Expected outcome 

> Over the course of the 10 week period, we expect the mentee to progress the prototype through web development and design and to work with autistic community members to implement search and moderation processes.  

Reviewer: ```suggestion
We expect that the GSoC 2021 mentee will progress the prototype AutSPACEs platform through web development and design. They will work with the AutSPACEs community members to implement search and moderation processes.  
```

### Required skills

> * Basic version control using GitHub
> * Python
> * Test-driven development

Reviewer: ```suggestion
* Version control using GitHub
* Python (intermediate-proficient)
* Basic understanding of test-driven development
```

* Skills or interests in UI/UX design
* Basic web development: ability to design and develop new web links and pages
* Clear, concise, and responsive communication and documentation
* Respect for, and the abiliity to work with, diverse collaborators.

### Optional skills

* Django
* Containers, docker
* Continuous integration
* Deployment and hosting
* Lived experience of autism

### Possible mentors

* Lead mentor: Georgia Aitkenhad (gaitkenhead@turing.ac.uk) 
* Co-mentor: James Kim (hyongsupkim@gmail.com) 

*No planned longer absences during the lead up/student interaction period (February 1 - April 19, especially the last week) or the project period (May 17 - August 30), however, it is possible based on the pandemic situation. 

> We have proposed two co-mentors so that at least one of us can be present to support the GSoC mentee at all time.*

Reviewer: ```suggestion
We have proposed co-mentors so that at least one of us can be present to support the GSoC mentee at all time.*
```

> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```

Reviewer: removing empty spaces
